cdi-core/src/main/java/com/linkedin/cdi/connection/S3Connection.java

```diff
@@ -72,6 +83,15 @@ public WorkUnitStatus execute(WorkUnitStatus status) {
 
     String finalPrefix = getWorkUnitSpecificString(s3SourceV2Keys.getPrefix(), getExtractorKeys().getDynamicParameters());
     LOG.debug("Final Prefix to get files list: {}", finalPrefix);
+    boolean shouldUpload = StringUtils.isNotEmpty(getExtractorKeys().getPayloadsBinaryPath());
```

Comment:
Seems very implicit. 
Well, If this is the only way.

Reply:
The intention here is to avoid adding extra parameters. That's why we choose to construct the flag this way